OffchainLabs/arbitrum-sdk
# feat: support old message delivered events

after v3 upgrade pre-nitro message don’t work anymore, so e.g. tx 0xc80e0c4844bb502ed7d7e2db6f9e6b2d52e3d25688de216394f0227fc5dc814e will fail if you try to call getL1ToL2Messages  on its receipt. Couple of points to mention:

- the code for calculating different values for v1 receipts was taken from sdk v1
- before the flow jumps to v1 calculation it checks for whether the current block is pre-nitro (is classic)
- currently working on improving types for the method returning v3 and classic messages

If you're interested in testing these changes locally.

_Prior to all steps below you'll have to set up the nitro node. Follow `Run integration tests` in https://github.com/OffchainLabs/arbitrum-sdk to set it up._

How to reproduce the issue on the main branch:

- create a new file somewhere, e.g. test.ts
- instead of hard-coding the entire txn like in the test, use the following code:

  `const provider = new JsonRpcProvider(YOUR_INFURA_KEY)`
`const arbProvider = new JsonRpcProvider('https://arb1.arbitrum.io/rpc')`
`const txRec = await provider.getTransactionReceipt('0xc80e0c4844bb502ed7d7e2db6f9e6b2d52e3d25688de216394f0227fc5dc814e')`
  `const l1rec = new L1TransactionReceipt(txRec)`
  `const msg = await l1rec.getL1ToL2Messages(arbProvider)`

- run the file: `ts-node ./test.ts`  - you might need to adjust the path to wherever you've put the file
  
  You should see an error saying that messages count doesn't match (or something along these lines)
